database.js

```diff
@@ -86,15 +87,22 @@ Database.prototype = {
       }
     }
 
-    return requestPromise(options)
-      .then(body => {
-        if (typeof body !== 'object') {
+    return request(options)
+      .then(response => {
+        if (typeof response.data !== 'object') {
           throw new Error('Received body was not an object')
         }
         if (this.cache && options.cacheKey) {
-          this.cache.set(options.cacheKey, body)
+          this.cache.set(options.cacheKey, response.data)
+        }
+        return response.data
+      })
+      .catch(err => {
+        if (err.response && err.response.data && err.response.data.message) {
+          throw new Error(err.response.data.message)
```

Comment:
I am not sure how smart this is as it includes loosing the error code returned by the request (404 and so on...) which can actually be really useful for end-users.